src/lib/expression/arithmetic_expression.cpp

```diff
@@ -0,0 +1,87 @@
+#include "arithmetic_expression.hpp"
+
+#include <sstream>
+
+#include "boost/functional/hash.hpp"
+#include "expression_utils.hpp"
+
+namespace opossum {
+
+std::ostream& operator<<(std::ostream& stream, const ArithmeticOperator arithmetic_operator) {
+  switch (arithmetic_operator) {
+    case ArithmeticOperator::Addition:
+      stream << "+";
+      break;
+    case ArithmeticOperator::Subtraction:
+      stream << "-";
+      break;
+    case ArithmeticOperator::Multiplication:
+      stream << "*";
+      break;
+    case ArithmeticOperator::Division:
+      stream << "/";
+      break;
+    case ArithmeticOperator::Modulo:
+      stream << "%";
+      break;
+    case ArithmeticOperator::Power:
+      stream << "^";
+      break;
+  }
+  return stream;
+}
+
+ArithmeticExpression::ArithmeticExpression(const ArithmeticOperator arithmetic_operator,
+                                           const std::shared_ptr<AbstractExpression>& left_operand,
+                                           const std::shared_ptr<AbstractExpression>& right_operand)
+    : AbstractExpression(ExpressionType::Arithmetic, {left_operand, right_operand}),
+      arithmetic_operator(arithmetic_operator) {}
+
+const std::shared_ptr<AbstractExpression>& ArithmeticExpression::left_operand() const { return arguments[0]; }
+
+const std::shared_ptr<AbstractExpression>& ArithmeticExpression::right_operand() const { return arguments[1]; }
+
+std::shared_ptr<AbstractExpression> ArithmeticExpression::deep_copy() const {
+  return std::make_shared<ArithmeticExpression>(arithmetic_operator, left_operand()->deep_copy(),
+                                                right_operand()->deep_copy());
+}
+
+DataType ArithmeticExpression::data_type() const {
+  return expression_common_type(left_operand()->data_type(), right_operand()->data_type());
+}
+
+std::string ArithmeticExpression::as_column_name() const {
+  std::stringstream stream;
+
+  stream << _enclose_argument_as_column_name(*left_operand()) << " " << arithmetic_operator << " "
+         << _enclose_argument_as_column_name(*right_operand());
+
+  return stream.str();
+}
+
+bool ArithmeticExpression::is_nullable() const {
+  return AbstractExpression::is_nullable() || arithmetic_operator == ArithmeticOperator::Division ||
```

Comment:
The `return AbstractExpression::is_nullable()` is a bit surprising and easy to forget in other expressions.

Reply:
Aggreed, but ArithmeticExpression really is the only place where `AbstractExpression::is_nullable()` this is needed.